projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -518,6 +528,37 @@ function wpcom_track_publish_first_post_task() {
 	wpcom_mark_launchpad_task_complete_if_active( 'first_post_published_newsletter' );
 }
 
+/**
+ * Callback for completing the Write posts task.
+ *
+ * @return void
+ */
+function wpcom_track_write_posts_task() {
+	// Ensure that Headstart posts don't mark this as complete
+	if ( defined( 'HEADSTART' ) && HEADSTART ) {
+		return;
+	}
+
+	$posts_count = wp_count_posts( 'post' );
+	if ( $posts_count->publish >= 3 ) {
+		wpcom_mark_launchpad_task_complete_if_active( 'write_posts' );
+	}
+}
+
+/**
+ * Callback for getting the number of posts published.
+ *
+ * @param array $task The Task definition.
+ * @return int
+ */
+function wpcom_get_write_posts_repetition_count( $task ) {
+	$published_posts_count = wp_count_posts( 'post' )->publish;
```

Comment:
We should create a shared function to compute the number of published, non-Headstart posts instead of relying on multiple calls to `wp_count_posts()`.